varnish-sys/src/vcl/convert.rs

```diff
@@ -267,13 +267,19 @@ impl IntoVCL<VCL_STRING> for &[u8] {
     fn into_vcl(self, ws: &mut Workspace) -> Result<VCL_STRING, VclError> {
         // Try to save some work if the buffer is already in the workspace.
         // We assume that &[u8] has always been readonly, so workspace data is valid.
-        if ws.contains(self) {
+        // However, we need to make sure that the buffer is null-terminated, either by its own last
+        // byte, or by the one after it
+        let with_extra_byte = unsafe { std::slice::from_raw_parts(self.as_ptr(), self.len() + 1) };
+        if (ws.contains(self) && *self.last().unwrap() == b'\0')
```

Comment:
rule of thumb - if you are doing `unwrap`, either it is wrong, or you need to have a comment explaining why it will never happen.  Especially when this is a result-returning function.  Empty slices are valid, so you may fail here because of that.

Reply:
correct, I missed that one

Reply:
completely forgot that we have destructuring in `rust`, I love that thing